### ISSUE TITLE
Handling EXTRUDE_MAXLENGTH

### ISSUE THREAD

**Issue by tablatronix** [jim-p/Change_Filament#15]
I am assuming this plugin is not working for me because my `EXTRUDE_MAXLENGTH` is 200mm

Can you detect the error Ill have to check exactly what it says,  it comes through terminal 

In settings allow extrude settings to be multiple moves, so 200mm x 2
Could be done with 1 extra input with a default of 1

And the code would have to handle this of course, and issue 2+ commands and calculate the time somehow ?( I mean we should be able to guess via the feedrate ) unless there is a way to ask in marlin for this settings and do it all automatically maybe....

**Comment by tablatronix** [comment]
refs

`#define EXTRUDE_MAXLENGTH (X_MAX_LENGTH+Y_MAX_LENGTH) //prevent extrusion of very large distances.`

`too long extrusion prevented`

**Comment by tablatronix** [comment]
it seems `M302` can disable something, looking into if it does both cold and lengthy flags in marlin.
I was thinking it disables `PREVENT_DANGEROUS_EXTRUDE` which covers both.

**Comment by jim-p** [comment]
The way this plugin is made, there isn't a way for it to read or interpret values returned from the printer. In your special case, you could set the unload/load length to half what you really want, and simply click the button twice.

If it becomes a more widespread issue I could maybe work around it some other way, like perhaps supporting comma-separated values in the fields and running multiple commands. For example, setting the unload length to `150,150` would result in a total of 300 issued as two commands.

**Comment by tablatronix** [comment]
Good idea

**Comment by jerryfudd** [comment]
Hi, I was coming onto report the same issue. could it be possible to set a multiplier in the plugin aswell as the extrude length?

**Comment by JakeShelf** [comment]
(Posting for content on this thread from #21)
The maximum I have managed to have my Ender 3 Pro unload is 100mm, and the full filament unload length is 475mm (slightly shorter Bowden tube fitted than the factory 500mm)

Is it possible to have the plugin feature filament unload length and maximum machine retract length, and then manage the retract from there with multiple commands?

**Comment by tablatronix** [comment]
But then you are asking the user to fill out 2 inputs, seems cumbersome, it would be cool if you could query marlin for `EXTRUDE_MAXLENGTH` but you cannot afaik

**Comment by JakeShelf** [comment]
Yes, two inputs, which are entered once and set and forget after that...

**Comment by quack3d** [comment]
I second this. 👍

**Comment by furyhh** [comment]
I'm new to this stuff, I just started editing marlin for my 3D printer, can someone explain to me what EXTRUDE MAXLENGTH means and how to set it up
